lms/djangoapps/certificates/signals.py

```diff
@@ -81,17 +82,40 @@ def _listen_for_id_verification_status_changed(sender, user, **kwargs):  # pylin
 
 def fire_ungenerated_certificate_task(user, course_key, expected_verification_status=None):
     """
-    Helper function to fire un-generated certificate tasks
-
-    The 'mode_is_verified' query is copied from the GeneratedCertificate model,
-    but is done here in an attempt to reduce traffic to the workers.
-    If the learner is verified and their cert has the 'unverified' status,
-    we regenerate the cert.
+    Helper function to fire certificate generation task.
+    Auto-generation of certificates is available for following course modes:
+        1- VERIFIED
+        2- CREDIT_MODE
+        3- PROFESSIONAL
+        4- NO_ID_PROFESSIONAL_MODE
+
+    Certificate generation task is fired to either generate a certificate
+    when there is no generated certificate for user in a particular course or
+    update a certificate if it has 'unverified' status.
+
+    Task is fired to attempt an update to a certificate
+    with 'unverified' status as this method is called when a user is
+    successfully verified, any certificate associated
+    with such user can now be verified.
+
+    NOTE: Purpose of restricting other course modes (HONOR and AUDIT) from auto-generation is to reduce
+    traffic to workers.
     """
+
+    allowed_enrollement_modes_list = [
+        CourseMode.VERIFIED,
+        CourseMode.CREDIT_MODE,
+        CourseMode.PROFESSIONAL,
+        CourseMode.NO_ID_PROFESSIONAL_MODE,
```

Comment:
I would add these modes to `GeneratedCertificates.VERIFIED_CERTS_MODES`. We probably don't want to depend on `CourseMode` in this file.

@iloveagent57 what do you think?

Reply:
It's fine to import `CourseMode` here, I don't think it will raise any dependency issue.  Also, as was mentioned somewhere in the corresponding ticket, `VERIFIED_CERTS_MODES` has some other logic associated with it that we don't want to change.  Here, we're just concerned with the conditions under which the `generate_certificate` task gets fired.

Reply:
Thanks for clarifying, @iloveagent57. Looks good then, @noraiz-anwar.